applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/explicit_strategy.hpp

```diff
@@ -0,0 +1,537 @@
+
+#if !defined(KRATOS_EXPLICIT_STRATEGY)
+#define  KRATOS_EXPLICIT_STRATEGY
+
+/* System includes */
+
+
+
+/* Project includes */
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "solving_strategies/strategies/solving_strategy.h"
+#include "structural_mechanics_application_variables.h"
+
+//default builder and solver
+#include "custom_strategies/custom_builder_and_solver/explicit_builder_and_solver.hpp"
+
+
+namespace Kratos
+{
+
+template<class TSparseSpace,
+         class TDenseSpace, // = DenseSpace<double>,
+         class TLinearSolver //= LinearSolver<TSparseSpace,TDenseSpace>
+         >
+class ExplicitStrategy
+    : public SolvingStrategy<TSparseSpace, TDenseSpace, TLinearSolver>
+{
+public:
+
+    /** Counted pointer of ClassName */
+
+    KRATOS_CLASS_POINTER_DEFINITION(ExplicitStrategy);
+
+    typedef SolvingStrategy<TSparseSpace, TDenseSpace, TLinearSolver> BaseType;
+
+    typedef typename BaseType::TBuilderAndSolverType TBuilderAndSolverType;
+
+    typedef typename BaseType::TDataType TDataType;
+
+    typedef TSparseSpace SparseSpaceType;
+
+    typedef typename BaseType::TSchemeType TSchemeType;
+
+    typedef typename BaseType::DofsArrayType DofsArrayType;
+
+    typedef typename BaseType::TSystemMatrixType TSystemMatrixType;
+
+    typedef typename BaseType::TSystemVectorType TSystemVectorType;
+
+    typedef typename BaseType::LocalSystemVectorType LocalSystemVectorType;
+
+    typedef typename BaseType::LocalSystemMatrixType LocalSystemMatrixType;
+
+    typedef typename BaseType::TSystemMatrixPointerType TSystemMatrixPointerType;
+
+    typedef typename BaseType::TSystemVectorPointerType TSystemVectorPointerType;
+
+    typedef ModelPart::NodesContainerType NodesArrayType;
+
+    typedef ModelPart::ElementsContainerType ElementsArrayType;
+
+    typedef ModelPart::ConditionsContainerType ConditionsArrayType;
+
+
+    /** Constructors.
+     */
+    ExplicitStrategy(
+        ModelPart& model_part,
+        bool MoveMeshFlag = true
+    )
+        : SolvingStrategy<TSparseSpace, TDenseSpace, TLinearSolver>(model_part, MoveMeshFlag)
+    {
+    }
+
+    ExplicitStrategy(
+        ModelPart& model_part,
+        typename TSchemeType::Pointer pScheme,
+        typename TLinearSolver::Pointer pNewLinearSolver,
+        bool CalculateReactions = false,
+        bool ReformDofSetAtEachStep = false,
+        bool MoveMeshFlag = true
+    )
+        : SolvingStrategy<TSparseSpace, TDenseSpace, TLinearSolver>(model_part, MoveMeshFlag)
+    {
+        KRATOS_TRY
+
+        //set flags to default values
+        mCalculateReactionsFlag = CalculateReactions;
+        mReformDofSetAtEachStep = ReformDofSetAtEachStep;
+
+	    //saving the scheme
+        mpScheme = pScheme;
+
+        //saving the linear solver
+        mpLinearSolver = pNewLinearSolver; //Not used in explicit strategies
+
+        //setting up the default builder and solver
+        mpBuilderAndSolver = typename TBuilderAndSolverType::Pointer
```

Comment:
Better to use a make_shared?

Reply:
take out the mpLinearSolver if you don¡t need it